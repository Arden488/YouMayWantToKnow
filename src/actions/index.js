import axios from 'axios';
import { YOUTUBE_API_KEY, FLICKR_API_KEY, NEWS_API_KEY } from '../config/api';

export function fetchTrends() {
  const request = axios.get('http://hawttrends.appspot.com/api/terms/');
  
  return (dispatch) => {
    dispatch({
      type: `TRENDS_FETCHING`,
    });

    request
      .then(response => {
        console.log(response);
        dispatch({
          type: `TRENDS_RECEIVED`,
          payload: response,
        });
      })
      .catch((err) => console.log('Error: ', err));
  };
}

export function fetchData(query) {
  return (dispatch) => {
    Promise.all([
      dispatch(fetchEach('ARTICLES', `https://newsapi.org/v2/everything?q=${query}&sortBy=popularity&apiKey=${NEWS_API_KEY}`)),
      dispatch(fetchEach('FLICKR', `https://api.flickr.com/services/rest/?api_key=${FLICKR_API_KEY}&method=flickr.photos.search&format=json&nojsoncallback=1&&per_page=9&page=1&text=${query}`)),
      dispatch(fetchEach('YOUTUBE', `https://www.googleapis.com/youtube/v3/search?part=id,snippet&q=${query}&type=video&key=${YOUTUBE_API_KEY}`)),
      dispatch(fetchEach('WIKI', `https://en.wikipedia.org/w/api.php?action=query&prop=pageimages|extracts&pithumbsize=300&format=json&origin=*&exintro=&exsentences=5&titles=${query}&redirects`)),
    ]).then(() => {
      dispatch({
        type: 'SET_QUERY',
        payload: query,
      });
    });
  }
}

export function fetchEach(name, url) {
  const request = axios.get(url);

  return (dispatch) => {
    dispatch({
      type: `${name}_FETCHING`,
    });

    request
      .then(response => {
        dispatch({
          type: `${name}_RECEIVED`,
          payload: response,
        });
      })
      .catch((err) => console.log('Error: ', err));
  };
}
