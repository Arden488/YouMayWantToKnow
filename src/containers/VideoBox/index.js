import React, { Component } from 'react';
import { connect } from 'react-redux';

import Grid from '@material-ui/core/Grid';
import List from '@material-ui/core/List';
import ListItem from '@material-ui/core/ListItem';
import ListItemAvatar from '@material-ui/core/ListItemAvatar';
import ListItemText from '@material-ui/core/ListItemText';
import Avatar from '@material-ui/core/Avatar';

class VideoBox extends Component {
  renderFirstVideo() {
    if (!this.props.videos || this.props.videos.length < 1)
      return <p>No video</p>;

    const videoId = this.props.videos[0].id.videoId;
    const url = `https://www.youtube.com/embed/${videoId}`;

    return <iframe src={url} width="100%" height="300" frameBorder="0" allow="autoplay; encrypted-media" allowFullScreen></iframe>;
  }

  renderOtherVideos() {
    if (!this.props.videos || this.props.videos.length < 2)
      return false;

    return (
      <List>
        {this.generateListItem(this.props.videos)}
      </List>
    )
  }

  generateListItem(data, element) {
    // delete data[Object.keys(data)[0]];
    // delete data[Object.keys(data)[data.length-1]];
    
    return data.slice(1,4).map(value =>
      <ListItem key={value.id.videoId} button>
        <img
          alt={value.snippet.title}
          src={value.snippet.thumbnails.default.url}
        />
        <ListItemText
          primary={value.snippet.title}
         />
      </ListItem>
    );
  }

  render() {
    return (
      <div>
        <Grid container spacing={24}>
          <Grid item xs={6}>
            {this.renderFirstVideo()}
          </Grid>
          <Grid item xs={6}>
            {this.renderOtherVideos()}
          </Grid>
        </Grid>
      </div>
    );
  }
}

function mapStateToProps(state) {
  return {
    videos: state.videos,
  }
}

export default connect(mapStateToProps)(VideoBox);