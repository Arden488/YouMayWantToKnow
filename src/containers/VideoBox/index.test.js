import React from 'react';
import VideoBox from './index';
import configureStore from 'redux-mock-store';
import { fetchYoutube } from '../../actions';
import reducerVideo from '../../reducers/reducer_video';
import thunk from 'redux-thunk';

describe('<VideoBox />', () => {
  let store;
  let component;
  let mockStore;

  beforeAll(() => {
    const middlewares = [ thunk ];
    mockStore = configureStore(middlewares);
    store = mockStore({});
    component = shallow(<VideoBox store={store} />);
  })

  it('renders correctly', () => {
    expect(component.dive()).toMatchSnapshot();
  });

  it('loads video', () => {
    store = mockStore({
      videos: [{ 'id': 'RS7IzU2VJIQ' }]
    });
    component = shallow(<VideoBox store={store} />);

    expect(component.dive()).toMatchSnapshot();
  });
});
