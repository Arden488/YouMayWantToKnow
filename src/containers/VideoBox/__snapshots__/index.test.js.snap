// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<VideoBox /> loads video 1`] = `
<div>
  <WithStyles(Grid)
    container={true}
    spacing={24}
  >
    <WithStyles(Grid)
      item={true}
      xs={6}
    >
      <iframe
        allow="autoplay; encrypted-media"
        allowFullScreen={true}
        frameBorder="0"
        height="300"
        src="https://www.youtube.com/embed/undefined"
        width="100%"
      />
    </WithStyles(Grid)>
    <WithStyles(Grid)
      item={true}
      xs={6}
    />
  </WithStyles(Grid)>
</div>
`;

exports[`<VideoBox /> renders correctly 1`] = `
<div>
  <WithStyles(Grid)
    container={true}
    spacing={24}
  >
    <WithStyles(Grid)
      item={true}
      xs={6}
    >
      <p>
        No video
      </p>
    </WithStyles(Grid)>
    <WithStyles(Grid)
      item={true}
      xs={6}
    />
  </WithStyles(Grid)>
</div>
`;
